sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/ControlFlowGraphTest.cs

```diff
@@ -5018,6 +5013,30 @@ public void Cfg_PropertyPatternClause_InsideIf()
             VerifyNoInstruction(exitBlock);
         }
 
+        [TestMethod]
+        [TestCategory("CFG")]
+        public void Cfg_DiscardPatternClause_InsideIf()
+        {
+            var cfg = Build(@"if (address is Address _) { return true; }");
+
+            VerifyCfg(cfg, 3);
+
+            var entryBlock = (BinaryBranchBlock)cfg.EntryBlock;
+            var trueBlock = (JumpBlock)cfg.Blocks.ElementAt(1);
+            var exitBlock = cfg.ExitBlock;
+
+            entryBlock.TrueSuccessorBlock.Should().Be(trueBlock);
+            entryBlock.FalseSuccessorBlock.Should().Be(exitBlock);
+            VerifyAllInstructions(entryBlock,
+                "address",
+                "address is Address _"); // IsPatternExpression
+
+            trueBlock.SuccessorBlock.Should().Be(exitBlock);
+            VerifyAllInstructions(trueBlock, "true");
+
+            VerifyNoInstruction(exitBlock);
+        }
+
```

Comment:
Is this related to the current PR?

Reply:
Nope. It remained from when I was testing several cases from the SE